### DISCUSSION TITLE
Managing AKS Welcome page on activation

### DISCUSSION THREAD

**Discussion by kejatura-dev** [Azure/vscode-aks-tools#1050] [General]
Hey all, 

Regarding this: https://github.com/microsoft/GitHub-Copilot-for-Azure/issues/13, it seems the welcome page is shown to the user every time the AKS extension is activated. With using @azure participant in GitHub Copilot, it needs to activate the AKS extension to find our handlers - causing unintended AKS welcome page to be displayed to the user whether or not the handler is chosen.

This only happens if:
- User has both GitHub Copilot for Azure + Azure Kubernetes Service extensions installed.
- User makes a request to @azure that pertains to one of the handlers written within AKS extension. 

Discussed with @Tatsinnit and we have come up with some proposed solutions.

Proposal: To reduce the noise from AKS welcome page, we can disable it:

1. 💡 **Disable AKS Welcome page** only if GitHub Copilot for Azure (ms-azuretools.vscode-azure-github-copilot) is installed.
    - This has been explored and tested: https://github.com/Azure/vscode-aks-tools/pull/1049

2. 💡 **Allow user to disable AKS Welcome page** by providing a notification with action buttons that allow user to decide if they want to forever hide welcome page, or remind them again in a couple weeks.
    - This will be similar to in https://github.com/Azure/vscode-aks-tools/compare/main...Tatsinnit:feature/gh-coilot-work-feature-flag?expand=1  

3. 💡 **Combination of both**

**Comment by hsubramanianaks** [comment]
Thanks @kejatura-dev for starting this discussion, I would vote for option# 02 where user can disable/enable this welcome page display via notification, that would fix it for all situations/scenarios instead of fixing it just for GH copilot extension. Please let me know your thoughts.

**Comment by joybb** [comment]
Few things I notice and would like us to improve the experience.
- I am working on my manifest files and once they were ready. I decided to visit the AKS extension to deploy the manifest. It on activation took my focus off my file and launched the landing page. 
- Activation page should follow common extension pattern, which is launch the screen on first installation and have it come up on upgrade. Other than that, if folks want, we can add a functionality to let user launch it.

I feel it's totally unnecessary to launch the landing page on every activation of the Extension.

**Comment by qpetraroia** [comment]
I like # 2 but I'm worried that it is more of a Band-Aid than a permanent solution. As @joybb mentioned, the launch page should be activated the first time a user installs the extension. After that it should go away. This bug may be from the past where we actually had a lot of issues getting the launch page to be activated.

If we are worried that users may not see certain commands/features, we could implement a new page that gets activated when an update is released which shows the release notes 
![image](https://github.com/user-attachments/assets/04e4f878-2aba-4eed-b4f2-c15b7e37bd65)

**Reply by joybb** [reply]
Do we have data on how many people use the walkthrough? This will also help in identifying if it's really being used.

**Comment by Tatsinnit** [comment]
### 💡  Key aspect for the Multi Pronged Approach we should consider for this as a long term fix.

👍 Thank you so much all, and chat ups, it was great seeing the actual usability issue which is happening.

So, instead of possible 1 bullet solution all this needs to be handled according to the unique condition and point of entry.

:shipit: This scenario is something I chatted with Joy and Kevin and worth to share. This also caters what Quentin mentioned above. 

#### 💡 🥷 **Scenario 1:** Auto Show Once Per Version Update/Release Walkthrough - To display a VS Code walkthrough only once per version.

In this case, we basically take care of the situation where the walkthrough only appears once per-version and then we store the value.

##### 🦑  Before you use vsix below to test this - Please be aware!!

* Upon **install** of the below VSIX - You will only ever get walkthrough once and then once mark-done or even closing it, it will not auto invoke until next version is released, obviously user can open the walkthrough on their own by `ctrl + shift + p` > `walkthroughs` and select the `AKS Walkthrough` 

For that I've made the change, and here is the vsix, rather than unzipping just remove the last .zip from the name and the vsix will be usable: 
[vscode-aks-tools-1.5.0-test-not-clause-5.vsix.zip](https://github.com/user-attachments/files/17674711/vscode-aks-tools-1.5.0-test-not-clause-5.vsix.zip)

Will update rest later... (thought At least share this once per install solution first)

**Reply by joybb** [reply]
Thanks, the vsix works great.

**Reply by kejatura-dev** [reply]
Just tested this too, it works. No longer seeing the walkthrough page after closing it

**Reply by Tatsinnit** [reply]
❤️ ❤️Implementation here:  https://github.com/Azure/vscode-aks-tools/pull/1051
